sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/component.py

```diff
@@ -115,7 +115,7 @@ def __init__(
             tags=tags,
             properties=properties,
             display_name=display_name,
-            is_deterministic=is_deterministic,
+            is_deterministic=is_deterministic,  # type: ignore[arg-type]
```

Comment:
Not sure why we would need to ignore here? Does the super class define this as a non-optional bool?

Reply:
Yes, below is the mypy error message:

Argument "is_deterministic" to "__init__" of "Component" has incompatible type "bool | None"; expected "bool"  [arg-type]